### ISSUE TITLE
Request Interrupts test failures

### ISSUE THREAD

**Issue by paul99** [v8mips/v8mips#134]
On mips64el, erpro-8, snapshots=on we get frequent failures on two cctests: 

```
cctest/test-api/RequestMultipleInterrupts
cctest/test-api/RequestInterruptTestWithNativeAccessor
```

Dusan mentioned these on this weeks v8 call as well.

Example builds:

```
https://v8.mips.com/bbv8/builders/Target%20-%20MIPS64%20-%20edge/builds/1205
https://v8.mips.com/bbv8/builders/Target%20-%20MIPS64%20-%20edge/builds/1201
https://v8.mips.com/bbv8/builders/Target%20-%20MIPS64%20-%20edge/builds/1198
```

Ákos, why don't you take a look at this one?

**Comment by paul99** [comment]
I just tested this today again, on ERpro-8 board, mips64el, snapshot on, and these errors still exist.

I can repro them on my board using build at today's rev:

```
commit 8c065681865599b29d52c01c6c644fa920c0b551
Author: arv <arv@chromium.org>
Date:   Thu Jun 4 14:16:18 2015 -0700
```

Built with mips-mti-linux-gnu-g++ (Codescape GNU Tools 2015.01-5 for MIPS MTI Linux) 4.9.2 from http://codescape-mips-sdk.imgtec.com/components/toolchain/2015.01-5/

I had both of these fail (plus several other tests) during a full 'check' run:

```
cctest/test-api/RequestMultipleInterrupts
cctest/test-api/RequestInterruptTestWithNativeAccessor
```

For me, they fail only when run under the test runner:

```
tools/run-tests.py -j2 --arch=mips64el --mode=release  --timeout=120 --verbose --time --no-presubmit --report cctest/test-api/RequestMultipleInterrupts
```

Using the above, it failed 1 test of the 4, in 2 runs out of 20. (18 runs passed all 4 tests)

It was always the `--stress-opt --always-opt` test that failed, but if I ran that test individually, it would not fail (in 50 tries)

RequestInterruptTestWithNativeAccessor failed more often, in 9 of 20 tries using test-runner, and in 2 of those failures 2 of the 4 tests failed. 

We would like to further reduce the test failure, to make it easier to reproduce, and of course, fix the bug.

Will assign to Ilija.

**Comment by dusmil** [comment]
Finally fixed in:
https://codereview.chromium.org/1220443002/

Some background details.

First Djordje proved my theory that infinite loop is finished before interrupt thread has been started because undefined value is occasionally returned in while condition. This also proves this issue is caused by the same bug  for issue 124. 

Second he proved that this was not the problem with interrupt handling mechanism by eliminating threading from the test.
Then several weeks, we came to the state where we add one nop and test was working and that was very indicative find.

**Comment by djpesic** [comment]
Fix commit id: 448ec36491a33111642470ad8b78e99ce3cf1975. Closing.
